File: src/main/java/preferences/logheaderdefinition/LogHeaderPreference.java
package preferences.logheaderdefinition;


import contract.Me7LogFileContract;
import io.reactivex.annotations.NonNull;

import java.util.prefs.BackingStoreException;
import java.util.prefs.Preferences;

public class LogHeaderPreference {

    private static final Preferences prefs = Preferences.userNodeForPackage(LogHeaderPreference.class);
    private final Me7LogFileContract.Header header;
    private final String defaultValue;

    public LogHeaderPreference(Me7LogFileContract.Header header, String defaultValue) {
        this.header = header;
        this.defaultValue = defaultValue;
        this.header.setHeader(getHeader());
    }

    public static void clear() {
        try {
            prefs.clear();
        } catch (BackingStoreException e) {
            e.printStackTrace();
        }
    }

    @NonNull
    public String getHeader() {
        return prefs.get(header.name(), defaultValue);
    }

    public void setHeader(@NonNull String headerLabel) {
        prefs.put(header.name(), headerLabel);
        header.setHeader(headerLabel);
    }
}

File: src/main/kotlin/data/preferences/logheaderdefinition/LogHeaderPreference.kt
package data.preferences.logheaderdefinition

import data.contract.Me7LogFileContract
import java.util.prefs.Preferences

object LogHeaderPreference {
    private val prefs = Preferences.userNodeForPackage(LogHeaderPreference::class.java)

    fun getHeader(header: Me7LogFileContract.Header): String {
        return prefs.get(header.name, header.header)
    }

    fun setHeader(header: Me7LogFileContract.Header, value: String) {
        prefs.put(header.name, value)
        header.header = value
    }

    fun loadHeaders() {
        for (header in Me7LogFileContract.Header.entries) {
            header.header = prefs.get(header.name, header.header)
        }
    }
}
